packages/product-json-to-csv/src/map-product-data.js

```diff
@@ -74,92 +106,257 @@ export default class ProductMapping {
+    })
+    return keywords
+  }
+
+  _mapVariantProperties(variant: Variant, productType: ?ProductType): Object {
+    const mappedVariant: Object = _.reduce(
+      variant,
+      (acc: Object, value: any, property: string): Object => {
+        acc[property] = this._mapVariantProperty(property, value)
+        return acc
+      },
+      {}
+    )
+
+    // complete missing attributes
+    if (productType && productType.attributes)
+      productType.attributes.forEach((attribute: Object) => {
+        const mappedAttributes = mappedVariant.attributes || {}
+        if (_.isNil(mappedAttributes[attribute.name]))
+          mappedAttributes[attribute.name] = ''
+      })
+
+    return mappedVariant
+  }
+
+  _mapVariantProperty(property: string, value: any): any {
+    switch (property) {
+      case 'attributes':
+        return this._mapAttributes(value)
+      case 'prices':
+        return this._mapPrices(value)
+      case 'images':
+        return this._mapImages(value)
+      case 'availability':
+        return value.availableQuantity
+      case 'id':
+      case 'sku':
+      case 'key':
+        return value
+      default:
+        return undefined
+    }
+  }
+
+  _mapPrices(prices: Array<Price>): string {
+    return prices
+      .map((price: Price) => ProductMapping._mapPrice(price))
+      .join(this.multiValDel)
+  }
+
+  static _mapPrice(price: Price): string {
+    // Full price:
+    // 'country-currencyCode centAmount|discounted.centAmount customerGroup.name#channel.key$validFrom~validUntil'
+    let priceString: string = ''
+
+    if (price.country) priceString += `${price.country}-`
+
+    priceString += `${price.value.currencyCode} ${price.value.centAmount}`
+
+    if (price.discounted) priceString += `|${price.discounted.value.centAmount}`
+
+    if (price.customerGroup && price.customerGroup.name)
+      priceString += ` ${price.customerGroup.name}`
+
+    if (price.channel && price.channel.key)
+      priceString += `#${price.channel.key}`
+
+    if (price.validFrom) priceString += `$${price.validFrom}`
+
+    if (price.validUntil) priceString += `~${price.validUntil}`
+
+    return priceString
+  }
+
+  _mapLtextAttribute(name: string, value: Object): Object {
+    // create an object with value based on selected language
+    const mappedAttribute = {
+      [name]: value[this.lang],
+    }
+    // create object with translations indexed with keys like: attributeName.en
+    const labels = flatten({
+      [name]: value, // value contains object with localized labels
+    })
+
+    // add labels in format "attribName.en: labelEnValue" to all attributes
+    return _.merge(mappedAttribute, labels)
+  }
+
+  _mapLabelsToAllAttributes(name: string, labels: Array<Object>): Object {
+    const mappedValues: Object = {}
+    const languages: Array<string> = _(labels)
+      .map(Object.keys)
+      .flatten()
+      .uniq()
+      .value()
+
+    languages.forEach(lang => {
```

Comment:
Here I don't know what is better - I know that updating the upper scope is not optimal but if we decide to use reduce, we would have to:
 - break `eslint` rule about not changing function parameters:
```js
return languages.reduce((mappedValues, lang) => {
  const headerKey: string = `${name}.${lang}`
  mappedValues[headerKey] = labels
    .map(label => label[lang] || '')
    .join(this.multiValDel)
  return mappedValues
}, {})

//  packages/product-json-to-csv/src/map-product-data.js:212:7
//  ✖  212:7  Assignment to property of function parameter mappedValues.  no-param-reassign
```
- or build always a new object created from a previous one enriched with the new key, which would have to copy always 1 attribute, then two, three, four.. so we would do a lot of clonning instead of reusing a single object.

WDYT? @adnasa @daern91